Indicators/BetaIndicator.cs

```diff
@@ -0,0 +1,182 @@
+ * limitations under the License.
+*/
+
+using System;
+using QuantConnect.Data.Market;
+using MathNet.Numerics.Statistics;
+
+namespace QuantConnect.Indicators
+{
+    /// <summary>
+    /// In technical analysis Beta indicator is used to measure volatility or risk of a target (ETF) relative to the overall 
+    /// risk (volatility) of the reference (market indexes). The Beta indicators compares target's price movement to the 
+    /// movements of the indexes over the same period of time.
+    /// 
+    /// It is common practice to use the SPX index as a benchmark of the overall reference market when it comes to Beta 
+    /// calculations.
+    /// </summary>
+    public class BetaIndicator : TradeBarIndicator, IIndicatorWarmUpPeriodProvider
+    {
+        /// <summary>
+        /// RollingWindow to store the data points of the target symbol
+        /// </summary>
+        private RollingWindow<decimal> _targetDataPoints;
+
+        /// <summary>
+        /// RollingWindow to store the data points of the reference symbol
+        /// </summary>
+        private RollingWindow<decimal> _referenceDataPoints;
+
+        /// <summary>
+        /// Symbol of the reference used
+        /// </summary>
+        private Symbol _referenceSymbol;
+
+        /// <summary>
+        /// Symbol of the target used
+        /// </summary>
+        private Symbol _targetSymbol;
+
+        /// <summary>
+        /// RollingWindow of returns of the target symbol in the given period
+        /// </summary>
+        private RollingWindow<double> _targetReturns;
+
+        /// <summary>
+        /// RollingWindow of returns of the reference symbol in the given period
+        /// </summary>
+        private RollingWindow<double> _referenceReturns;
+
+        /// <summary>
+        /// Beta of the target used in relation with the reference
+        /// </summary>
+        private decimal _beta;
+
+        /// <summary>
+        /// Required period, in data points, for the indicator to be ready and fully initialized.
+        /// </summary>
+        public int WarmUpPeriod { get; private set; }
+
+        /// <summary>
+        /// Gets a flag indicating when the indicator is ready and fully initialized
+        /// </summary>
+        public override bool IsReady => _targetDataPoints.Samples >= WarmUpPeriod && _referenceDataPoints.Samples >= WarmUpPeriod;
+
+        /// <summary>
+        /// Creates a new BetaIndicator with the specified name, period, target and 
+        /// reference values
+        /// </summary>
+        /// <param name="name">The name of this indicator</param>
+        /// <param name="period">The period of this indicator</param>
+        /// <param name="targetSymbol">The target symbol of this indicator</param>
+        /// <param name="referenceSymbol">The reference symbol of this indicator</param>
+        public BetaIndicator(string name, int period, Symbol targetSymbol, Symbol referenceSymbol)
```

Comment:
Mind renaming this to be inline with the other indicators? e.g. `ExponentialMovingAverage.cs` / so `Beta.cs`

Reply:
But of course. No problem!